### PR TITLE
"latest" to "nightly"

### PR DESCRIPTION
**Pull request by mccalluc**
- Addresses ["More generally, all docs.opendp.org/en/latest/ links are now broken."](https://github.com/opendp/opendp/issues/904#issuecomment-1667025539)
- Extends #925
- Also changes references to `latest` branch to `nightly`

Does not:
- touch `new-contribution.md`, so you can avoid merge conflicts
- touch any of the `https://docs.rs/opendp/latest/opendp/...` references in Python docstrings: These still work, and changing them to `nightly` doesn't, but that might also be a bug in the documentation generation process?

The https://github.com/opendp/opendp/tree/latest branch could probably be deleted on github to reduce confusion?

### CODE REVIEW

File: docs/source/contributor/maintainer-notes.rst [hunk 1]
```diff
@@ -180,10 +180,10 @@ These are the GitHub workflows that support the release process.
 sync-branches.yml
 ^^^^^^^^^^^^^^^^^
 
-* Keeps the tracking branches ``latest`` and ``stable`` in sync with their targets.
+* Keeps the tracking branches ``nightly`` and ``stable`` in sync with their targets.
   This is used when generating docs, so that we have a consistent path to each category.
 * Triggered on every push to ``main``, or when release is published.
-* Whenever there's a push to ``main``, it advances ``latest`` to the same ref.
+* Whenever there's a push to ``main``, it advances ``nightly`` to the same ref.
```
Comment (andrewvyrros) [inline docs/source/contributor/maintainer-notes.rst:186]: This whole doc is out of date with the new release process, filed #927.

File: docs/source/user/getting-started.ipynb [hunk 2]
```diff
@@ -344,7 +344,7 @@
       "    \n",
       "    **Proof Definition:**\n",
       "    \n",
-      "    [(Proof Document)](https://docs.opendp.org/en/latest/proofs/rust/src/transformations/clamp/make_clamp.pdf)\n",
+      "    [(Proof Document)](https://docs.opendp.org/en/nightly/proofs/rust/src/transformations/clamp/make_clamp.pdf)\n",
```
Comment (andrewvyrros) [inline docs/source/user/getting-started.ipynb:347]: Probably better to reference `stable` here.

File: python/src/opendp/_lib.py [hunk 3]
```diff
@@ -209,7 +209,7 @@ def versioned(function):
 
         docs_ref = get_docs_ref(version)
         function.__doc__ = function.__doc__.replace(
-            "https://docs.opendp.org/en/latest/",
+            "https://docs.opendp.org/en/nightly/",
```
Comment (andrewvyrros) [inline python/src/opendp/_lib.py:212]: This is as actually as intended, the codegen in the `opendp_tooling` crate creates the link using `latest`. (We're going to clean this up.)

File: python/src/opendp/accuracy.py
Comment (andrewvyrros) [inline python/src/opendp/accuracy.py:1]: This is a generated file, is as expected by the `@versioned` decorator.

File: python/src/opendp/measurements.py
Comment (andrewvyrros) [inline python/src/opendp/measurements.py:1]: This is a generated file, is as expected by the `@versioned` decorator.

File: python/src/opendp/transformations.py
Comment (andrewvyrros) [inline python/src/opendp/transformations.py:1]: This is a generated file, is as expected by the `@versioned` decorator.

**Review by andrewvyrros — COMMENTED** [review]
Thanks for finding these! Note that some of this is as intended, and the whole thing should be cleaned up with #928.

File: docs/source/contributor/maintainer-notes.rst (hunk 1 above)
Comment (mccalluc) [reply docs/source/contributor/maintainer-notes.rst:186]: I'll revert this then, so it doesn't conflict with other work.

**Comment by mccalluc** [comment]
I think latest commits address the comments... but if it all will be covered by #928 it's fine to close this without merging. Whatever is best.

**Comment by raprasad** [comment]
Closing for now.

**Pull request closed without merging**